jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -104,11 +130,35 @@ def _parse_image(app, file_obj, data_label, show_in_viewer, ext=None):
         raise NotImplementedError(f'Imviz does not support {file_obj}')
 
     for data, data_label in data_iter:
+
+        # avoid duplicate data labels in colection
+        if data_label in app.data_collection.labels:
+            data_label = data_label + "_2"  # 0th-order solution as proposed in issue #600
```

Comment:
Ah... I guess "append to infinity" is a simpler solution than try to keep the count. In this case, does not have to be `"_2"` since the number 2 appears arbitrary.

However, this also introduces a "bug" in the workflow. Let's consider the case where I ran a cell to load `myfile.fits[SCI]` into Imviz. I did something I didn't like. I ran the cell again to overwrite what I loaded, expecting the data label to remain the same. But now, I would see a new Data loaded with original label + suffix.

How do we differentiate the need between "reload" and "loading new data with clashing label"? I think this needs more discussion; that is why I advised to not implement this part yet until we can think about it more.

Reply:
I left this in place, since I understand that @eteq would like to see a few cases of real usage of this before we decide to either remove it, or make it smarter.

Reply:
Hmm.  @pllim makes a great point here. I honestly am not sure offhand which is the most natural solution.  But that makes me think maybe we should do as @ibusko said and just leave this in as the simplest to implement?  But should definitely add in a `TODO` pointing out @pllim's failure mode here, and add it to the issue tracker! That way the rest of this can go in while we work that part out?